src/modules/CRUD/services/CRUDProvider.js

```diff
@@ -29,105 +33,185 @@ export class CRUDProvider extends React.Component {
+      store,
+      error,
+      ids,
+    } = props;
+    const { errors: { [store]: errorStore } } = this.state;
+
+    const propsFromIds = Object.keys(ids);
+
+    // Remove error if it was before in the store
+    if (!(error instanceof Error)) {
+      return errorStore.filter(item => (
+        propsFromIds.every(prop => item[prop] !== ids[prop])
+      ));
     }
+
+
+    // Modify error if it was before in the store
+    const isErrorItemAlreadyExisting = errorStore.some(item => (
+      propsFromIds.every(prop => item[prop] === ids[prop])
+    ));
+
+    if (isErrorItemAlreadyExisting) {
+      return errorStore.map(item => (
+        propsFromIds.every(prop => item[prop] === ids[prop])
+          ? { error, ...ids }
+          : item
+      ));
+    }
+
+    // Else add error in the store
+    return [...errorStore, { error, ...ids }];
   }
 
   getFeaturesList = async layerId => {
+    const result = {};
     try {
-      const { results: featuresList = [] } = await fetchFeaturesList(layerId);
-      this.setState({ featuresList });
+      const { results: featuresList } = await fetchFeaturesList(layerId);
+      result.featuresList = featuresList;
     } catch (e) {
-      this.setState(state => ({
-        ...state,
-        errors: { ...state.errors, [state.featuresList.length]: true },
-      }));
+      result.error = e;
     }
-  }
+    const { featuresList = [], error = {} } = result;
+    const { errors } = this.state;
+
+    this.setState({
+      featuresList,
+      errors: {
+        ...errors,
+        featuresList: this.getFormattedError({ error, ids: { layerId }, store: 'featuresList' }),
+      },
+    });
+    return featuresList.length && featuresList;
+  };
 
   fetchFeature = async (layerId, featureId) => {
+    const result = {};
     try {
       const feature = await fetchFeatureAction(layerId, featureId);
-      this.setState(state => ({
-        feature: {
-          ...state.feature,
-          [feature.identifier]: feature,
-        },
-      }));
+      result.feature = feature;
     } catch (e) {
-      this.setState(state => ({
-        ...state,
-        errors: { ...state.errors, [featureId]: true, code: e.message },
-      }));
+      result.error = e;
     }
+
+    const { feature = {}, error = {} } = result;
+    const { featuresList: prevFeaturesList } = this.state;
+    const isFeatureAlreadyExisting = prevFeaturesList.some(({ identifier }) => (
+      identifier === feature.identifier
+    ));
+
+
+    const featuresList = isFeatureAlreadyExisting
+      ? prevFeaturesList.map(item => (item.identifier === feature.identifier ? feature : item))
+      : [...prevFeaturesList, feature.identifier && feature].filter(Boolean);
+
+    this.setState(({ errors }) => ({
+      featuresList,
+      errors: {
+        ...errors,
+        feature: this.getFormattedError({ error, ids: { layerId, featureId }, store: 'feature' }),
+      },
+    }));
+
+    return Object.keys(feature).length && feature;
```

Comment:
If length is `0`, `fetchFeature` will `return 0;`. Is that the right behavior?

Reply:
Right, in this case the expected value is `false`. I fix it.

Reply:
`!!` could have been the solution, but ok with yours.